### PR TITLE
Add a method to get the current orientation of the StatusBar [iOS only]

### PR DESCRIPTION
**Pull request by eduardinni**
This new method for StatusBar uses the native way to get the current orientation of the app's status bar:
```objc
[[UIApplication sharedApplication] statusBarOrientation]
```
This is useful to get device orientation easily, or as a fallback method when other device orientation techniques fail.

Example usage:
```js
import { StatusBar } from 'react-native';
StatusBar.getStatusBarOrientation(orientation => console.log(orientation));
```

### CODE REVIEW

**Comment by mkonicek** [comment]
Thanks for adding it! Please use `Promise`s instead of `callbacks`. Look at other native modules, for example: https://github.com/facebook/react-native/blob/9ee815f6b52e0c2417c04e5a05e1e31df26daed2/Libraries/Linking/Linking.js

**Comment by eduardinni** [comment]
Thanks for the review @mkonicek, I changed callbacks to promises in this new commit

File: React/Modules/RCTStatusBarManager.m [hunk 1]
```diff
@@ -103,6 +103,26 @@ - (void)applicationWillChangeStatusBarFrame:(NSNotification *)notification
   }]);
 }
 
+RCT_EXPORT_METHOD(getStatusBarOrientation:(RCTPromiseResolveBlock)resolve
+                                   reject:(RCTPromiseRejectBlock)reject)
+{
+  NSDictionary *orientations = @{
```
Comment (shergin) [inline React/Modules/RCTStatusBarManager.m]: `RCTConvert`?

**Comment by eduardinni** [comment]
@shergin, thanks for the review. Only the 'setter' of `statusBarOrientation` is deprecated since iOS 9 but the getter is working ok without warnings up to 10.3. `interfaceOrientation` is deprecated since iOS 8. 

`statusBarOrientation` returns the UIInterfaceOrientation enum. And the main intention to expose this method is to use as a fallback method when other device orientation 'getters' failed (like accelerometer without movement). I used it to deal with video recording orientation and rotation.

**Comment by shergin** [comment]
Yeah, seems you are right about `setStatusBarOrientation`.
```
// Explicit setting of the status bar orientation is more limited in iOS 6.0 and later.
@property(readwrite, nonatomic) UIInterfaceOrientation statusBarOrientation NS_DEPRECATED_IOS(2_0, 9_0, "Explicit setting of the status bar orientation is more limited in iOS 6.0 and later") __TVOS_PROHIBITED;
- (void)setStatusBarOrientation:(UIInterfaceOrientation)interfaceOrientation animated:(BOOL)animated NS_DEPRECATED_IOS(2_0, 9_0, "Explicit setting of the status bar orientation is more limited in iOS 6.0 and later") __TVOS_PROHIBITED;
```
So, you are using `[[UIDevice currentDevice] orientation]` right now and sometimes it fails, and in this case you want to rely on `statusBarOrientation`?

If so, I would propose to change the name to `orientation` or `interfaceOrientation` and use `RCTConvert` for converting between enum values and strings (https://github.com/facebook/react-native/blob/e1577df1fd70049ce7f288f91f6e2b18d512ff4d/React/Views/RCTScrollViewManager.m#L25).

**Comment by eduardinni** [comment]
Yeah, I found this answer and it explains good the difference:
https://stackoverflow.com/a/27830166

> If you want to get device orientation directly from accelerometer use [[UIDevice currentDevice] orientation]. But if you need current orientation of your application(interface orientation) use [[UIApplication sharedApplication] statusBarOrientation].

I will modify the code using `RCTConvert` and changing the name.

**Comment by hramos** [comment]
Any interest in getting this merged? Looks like it has been in the needs-revision state for two months now.

File: React/Modules/RCTStatusBarManager.m (hunk 1 above)
Comment (eduardinni) [inline React/Modules/RCTStatusBarManager.m]: Hello, I'm having troubles to invoque RCTConvert in the returned value, can you help me, please?

The setup:
```
RCT_ENUM_CONVERTER(UIInterfaceOrientation, (@{
  @"portrait": @(UIInterfaceOrientationPortrait),
  @"portrait-upside-down": @(UIInterfaceOrientationPortraitUpsideDown),
  @"landscape-left": @(UIInterfaceOrientationLandscapeLeft),
  @"landscape-Right": @(UIInterfaceOrientationLandscapeRight),
}), UIInterfaceOrientationUnknown, integerValue);
```

```
RCT_EXPORT_METHOD(getInterfaceOrientation:(RCTResponseSenderBlock)callback)
{
  callback(@[@{
    @"orientation": @([[UIApplication sharedApplication] statusBarOrientation]),
  }]);
}
```

I've seen many examples inside the source code of other components where converting input args, but none for converting returned values, how can I invoque the converter? For example:

Convert `UIInterfaceOrientationPortrait` --> `"portrait"`

Thanks

Comment (shergin) [inline React/Modules/RCTStatusBarManager.m]: You probably have to convert the value manually using RCTConvert's API.

Comment (eduardinni) [inline React/Modules/RCTStatusBarManager.m]: Thanks. I think RCTConvert is only helpful for setter methods. I could't use it to _reverse_ convert without re-declaring a dictionary.

**Comment by eduardinni** [comment]
I think RCTConvert is only helpful for setter methods. I couldn't use it to _reverse_ convert without re-declaring a dictionary. But I added constants to StatusBar component (like in `Image.resizeMode`) to make comparisons more clearer.

File: Libraries/Components/StatusBar/StatusBar.js [hunk 2]
```diff
@@ -249,6 +250,19 @@ class StatusBar extends React.Component {
...
+   */
+  static getOrientation(): Promise<?string> {
+    if (Platform.OS !== 'ios') {
+      console.warn('`getOrientation` is only available on iOS');
+      return;
+    }
+    return StatusBarManager.getOrientation();
+  }
+
+  static orientation = StatusBarOrientation;
```
Comment (shergin) [inline Libraries/Components/StatusBar/StatusBar.js]: Why do we need this?

File: Libraries/Components/StatusBar/StatusBar.js [hunk 3]
```diff
@@ -249,6 +250,19 @@ class StatusBar extends React.Component {
     StatusBarManager.setTranslucent(translucent);
   }
 
+  /**
+   * Get the current orientation of the status bar (portrait, portraitUpsideDown, landscapeLeft, landscapeRight)
+   */
+  static getOrientation(): Promise<?string> {
+    if (Platform.OS !== 'ios') {
+      console.warn('`getOrientation` is only available on iOS');
+      return;
```
Comment (shergin) [inline Libraries/Components/StatusBar/StatusBar.js]: Should we reject promise here?

**Pull request closed without merging**